js/signaling.js

```diff
@@ -192,13 +195,43 @@
 					}.bind(this));
 					callback('', callDescription);
 				}.bind(this));
+			}.bind(this),
+			error: function (result) {
+				if (result.status === 404 || result.status === 503) {
+					// Room not found or maintenance mode
+					OC.redirect(OC.generateUrl('apps/spreed'))
+				}
+
+				if (result.status === 403) {
+					// Invalid password
+					OC.dialogs.prompt(
+						t('spreed', 'Please enter the password for this call'),
```

Comment:
Actually it would be nice to have the room object in here so we can show the name, if that is okay to leak...